### ISSUE TITLE
22.7.2-amd64 - igmpproxy crashes

### ISSUE THREAD

**Issue by firefart** [opnsense/ports#158]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

Since the update to 22.7.2-amd64 (including kernel) the igmpproxy crashes with the following error. Before the update (22.7.1) everything was running without issues for months. Sadly there is no much information in the logs it seems, it just crashes. The crash also only kills the functionality of igmpproxy, the service is still running and opnsense reports it as running too

```
2022-08-18T23:58:54 | Warning | igmpproxy | select() failure; Errno(4): Interrupted system call
```

**Additional context**

config:

```

##------------------------------------------------------
## Enable Quickleave mode (Sends Leave instantly)
##------------------------------------------------------
quickleave
phyint igb0 upstream ratelimit 0 threshold 10
altnet 239.0.0.0/8
altnet 224.0.0.0/4
altnet 213.33.0.0/16

phyint igb1_vlan3 downstream ratelimit 0 threshold 1

phyint igb1_vlan4 disabled
phyint igb1_vlan5 disabled
phyint igb1 disabled
phyint ovpnc3 disabled
phyint igb1_vlan6 disabled
```

Source code line:
https://github.com/pali/igmpproxy/blob/0e7186b300c063ef4015f1551100765ef5537d4c/src/igmpproxy.c#L347

The error seems to be ignored but the cause seems to kill igmpproxy

**Environment***

```
| Type | opnsense |  
| Version | 22.7.2 |  
| Architecture | amd64 |  
| Flavour | OpenSSL |  
| Commit | 412c0b79c |  
| Mirror | https://pkg.opnsense.org/FreeBSD:13:amd64/22.7 |  
| Repositories | OPNsense |  
| Updated on | Thu Aug 18 23:54:00 CEST 2022 |  
| Checked on | N/A
```

**Comment by fichtner** [comment]
@firefart I'd be happy to look closer but the error is ignored properly (can always happen with system calls) and thus we are missing the real cause for the exit. Check dmesg and a complete log would be helpful.

**Comment by firefart** [comment]
@fichtner thanks! dmesg did not show anything unusual and running igmpproxy in debug also did not yield any hints to where the error lies. I also did several tcpdumps but also nothing from interest showing up. I think that it might be a kernel change that messes things up in my setup as igmpproxy did not change in this release and the rest of the packages where php related ones iirc.
For now I changed my IPTV setup to not require igmpproxy by connecting it directly to the internet
